src/middlewared/middlewared/plugins/crypto.py

```diff
@@ -164,9 +170,129 @@ class CryptoKeyService(Service):
         'email_address': 'email'
     }
 
+    EXTENSIONS = {}
+
     class Config:
         private = True
 
+    @staticmethod
+    def extensions():
+        if not CryptoKeyService.EXTENSIONS:
+            # For now we only support the following extensions
+            # We also support SubjectAlternativeName but as we include that natively if the user provides it
+            # we don't expose it to the end user as an extension making the process for the end user easier to
+            # create a certificate/ca as most wouldn't even want to know what extension is or does.
+            # Apart from this we also add subjectKeyIdentifier automatically
+            supported = [
+                'BasicConstraints', 'AuthorityKeyIdentifier',
+                'ExtendedKeyUsage', 'KeyUsage'
+            ]
+
+            for attr in filter(
+                lambda attr: attr in supported, dir(x509.extensions)
+            ):
```

Comment:
This seems a bit backwards to me, why loop through all x509 extensions if we already know the extensions we support and could just do `hasattr(attr, x509.extensions) for attr in supported`?